explainer.md

```diff
@@ -6,58 +6,191 @@
+    buffer = await response.arrayBuffer();
+    const biasData119 = new Float32Array(buffer, 0, BIAS_SIZE);
+    const biasData139 = new Float32Array(buffer, BIAS_SIZE, BIAS_SIZE);
+    const biasData159 = new Float32Array(buffer, 2 * BIAS_SIZE, BIAS_SIZE);
+    response = await fetch(initializerUrl);
+    buffer = await response.arrayBuffer();
+    const initData160 = new Float32Array(buffer, 0, MATMUL96_INIT_SIZE);
+    const initData170 = new Float32Array(buffer, MATMUL96_INIT_SIZE, ADD97_INIT_SIZE);
+    // Create constant operands
+    const builder = navigator.ml.getNeuralNetworkContext().creatModelBuilder();
+    const hidden1 = builder.constant({ type: 'float32', dimensions: HIDDEN_DIMS }, hiddenData1);
+    const hidden2 = builder.constant({ type: 'float32', dimensions: HIDDEN_DIMS }, hiddenData2);
+    const hidden3 = builder.constant({ type: 'float32', dimensions: HIDDEN_DIMS }, hiddenData3);
+    const weight117 = builder.constant({ type: 'float32', dimensions: WEIGHT_DIMS }, weightData117);
+    const weight118 = builder.constant({ type: 'float32', dimensions: WEIGHT_DIMS }, weightData118);
+    const weight137 = builder.constant({ type: 'float32', dimensions: WEIGHT_DIMS }, weightData137);
+    const weight138 = builder.constant({ type: 'float32', dimensions: WEIGHT_DIMS }, weightData138);
+    const weight157 = builder.constant({ type: 'float32', dimensions: WEIGHT_DIMS }, weightData157);
+    const weight158 = builder.constant({ type: 'float32', dimensions: WEIGHT_DIMS }, weightData158);
+    const bias119 = builder.constant({ type: 'float32', dimensions: BIAS_DIMS }, biasData119);
+    const bias139 = builder.constant({ type: 'float32', dimensions: BIAS_DIMS }, biasData139);
+    const bias159 = builder.constant({ type: 'float32', dimensions: BIAS_DIMS }, biasData159);
+    const init160 = builder.constant({ type: 'float32', dimensions: MATMUL96_INIT_DIMS }, initData160);
+    const init170 = builder.constant({ type: 'float32', dimensions: ADD97_INIT_DIMS }, initData170);
+    // Build up the network
+    const input = builder.input('input', { type:'float32', dimensions:INPUT_DIMS });
+    const [gru43, gru42] = builder.gru(input, weight117, weight118, STEP_COUNT, 257, 
+                                       bias119, null, hidden1, true, true);
+    const add45 = builder.add(input, builder.squeeze(gru42, 1));
+    const [gru68, gru67] = builder.gru(add45, weight137, weight138, STEP_COUNT, 257, 
+                                       bias139, null, hidden2, true, true);
+    const add70 = builder.add(add45, builder.squeeze(gru67, 1));
+    const [gru93, gru92] = builder.gru(add70, weight157, weight158, STEP_COUNT, 257, 
+                                       bias159, null, hidden3, true, true);
+    const output = builder.clamp(
+                    builder.sigmoid(
+                        builder.add(builder.matmul(builder.squeeze(gru92, 1), init160), init170)
+                        ), 0);
```

Comment:
WebNN [clamp](https://webmachinelearning.github.io/webnn/#api-modelbuilder-clamp) requires specifying the `maxValue`. Use `relu` here instead?

Reply:
I think both the min and the max value of `clamp` should be made optional. In this use case, the max value is left unspecified which should mean float_max.

Reply:
it makes sense. I'll send a PR to change the minValue and maxValue of clamp to optional.

Reply:
Another thing is today's clamp accepts minValue and maxValue as Operand. So it should be 
```js
builder.clamp(x, builder.constant(0));
```

Reply:
> I think both the min and the max value of `clamp` should be made optional. In this use case, the max value is left unspecified which should mean float_max.

Sent #110 to fix it. Please take a look. Thanks!

Reply:
I'm not entirely sure why `min` and `max` need to be operands and not just floats.

Reply:
Fixed.